src/operator/journal.rs

```diff
@@ -63,34 +48,38 @@ async fn reconcile(journal: Journal, ctx: Context<Data>) -> Result<ReconcilerAct
     let recorder = Recorder::new(client.clone(), reporter, journal.object_ref(&()));
     let name = ResourceExt::name(&journal);
     let ns = ResourceExt::namespace(&journal).expect("journal is namespaced");
-    let journals: Api<Journal> = Api::namespaced(client, &ns);
+    let journals: Api<Journal> = Api::namespaced(client.clone(), &ns);
+    let deploys: Api<Deployment> = Api::namespaced(client.clone(), &ns);
 
     let new_status = Patch::Apply(json!({
         "apiVersion": "engula.io/v1alpha1",
         "kind": "Journal",
         "status": JournalStatus {
-            is_bad: journal.spec.info.contains("bad"),
-            //last_updated: Some(Utc::now()),
+            deployment_status: None,
         }
     }));
     let ps = PatchParams::apply("cntrlr").force();
     let _o = journals
         .patch_status(&name, &ps, &new_status)
         .await
         .map_err(Error::KubeError)?;
-
-    if journal.spec.info.contains("bad") {
-        recorder
-            .publish(Event {
-                type_: EventType::Normal,
-                reason: "BadJournal".into(),
-                note: Some(format!("Sending `{}` to detention", name)),
-                action: "Correcting".into(),
-                secondary: None,
-            })
-            .await
-            .map_err(Error::KubeError)?;
-    }
+    let current = match deploys.get(&name).await {
+        Ok(current) => Some(current),
+        Err(kube::KubeError::Api(e)) => None,
```

Comment:
1. replace `kube::KubeError::Api(e)` with `kube::Error::Api(e)`
2. pattern matching requires a full match, therefore we should consider other candidate of `kube::Error` instead of just `kube::KubeError::Api(e)`

```rust
    let current = match deploys.get(&name).await {
        Ok(current) => Some(current),
        Err(kube::Error::Api(e)) => None,
        _ => None // should actually throw the err
```

3. this will make Option type here insufficient:
    1. OK path
    2. Not found path
    3. actual error path <- consider use another sum type to do this

Reply:
make current a `Result<Option<Deployment>, Error>` maybe a better choice